osbenchmark/aggregator.py

```diff
@@ -0,0 +1,216 @@
+        weighted_metrics = {}
+
+        for metric, values in task_metrics.items():
+            weighted_metrics[metric] = {}
+            if isinstance(values[0], dict):
+                for item_key in values[0].keys():
+                    if item_key == 'unit':
+                        weighted_metrics[metric][item_key] = values[0][item_key]
+                    else:
+                        item_values = [value.get(item_key, 0) for value in values]
+                        if iterations > 1:
+                            weighted_sum = sum(value * iterations for value in item_values)
+                            total_iterations = iterations * len(values)
+                            weighted_metrics[metric][item_key] = weighted_sum / total_iterations
+                        else:
+                            weighted_metrics[metric][item_key] = sum(item_values) / len(item_values)
+            else:
+                if iterations > 1:
+                    weighted_sum = sum(value * iterations for value in values)
+                    total_iterations = iterations * len(values)
+                    weighted_metrics[metric] = weighted_sum / total_iterations
+                else:
+                    weighted_metrics[metric] = sum(values) / len(values)
+        return weighted_metrics
+
+    # verify that all test executions have the same workload
+    def compatibility_check(self, test_store) -> None:
+        first_test_execution = test_store.find_by_test_execution_id(list(self.test_executions.keys())[0])
+        workload = first_test_execution.workload
+        for id in self.test_executions.keys():
+            test_execution = test_store.find_by_test_execution_id(id)
+            if test_execution:
+                if test_execution.workload != workload:
+                    raise ValueError(f"Incompatible workload: test {id} has workload '{test_execution.workload}' instead of '{workload}'")
+            else:
+                raise ValueError("Test execution not found: ", id)
+        return True
+
+    # driver code
+    def aggregate(self) -> None:
+        test_execution_store = metrics.test_execution_store(self.config)
+        if self.compatibility_check(test_execution_store):
+            for id in self.test_executions.keys():
+                test_execution = test_execution_store.find_by_test_execution_id(id)
+                if test_execution:
+                    self.config.add(config.Scope.applicationOverride, "workload", "repository.name", "default")
+                    self.config.add(config.Scope.applicationOverride, "workload", "workload.name", test_execution.workload)
+                    self.iterations()
+                    self.results(test_execution)
+
+            aggregated_results = self.build_aggregated_results(test_execution_store)
+            file_test_exe_store = FileTestExecutionStore(self.config)
+            file_test_exe_store.store_test_execution(aggregated_results)
```

Comment:
A couple of questions:
- Will we be storing the aggregated results to ~/.benchmark/benchmarks/test_executions or to a separate directory for aggregated results?
- Will we also store this in a OSTestExecutionStore if the user has their benchmark.ini file configured to use an external metrics data store? If so, let's implement this in a separate PR

Reply:
For now, we'll store the results to the benchmarks test_executions folder, but I can add a separate folder in a future PR.

I did some testing and this does store in an OSTestExecutionStore when my benchmark.ini file is configured to use it!

Reply:
Awesome, sounds good!